lib/dialects/abstract/connection-manager.js

```diff
@@ -98,7 +101,10 @@ class ConnectionManager {
 
     if (!config.replication) {
       this.pool = Pooling.createPool({
-        create: () => this._connect(config).catch(err => err),
+        create: () =>
+          retry(() => this._connect(config), {
```

Comment:
This is not fixing any issue, we are just retrying 5 more time. Real issue is still there

Reply:
I agree, when I was reading the codes that's what I thought as well. However, the weird thing is, a few others mentioned that that PR managed to solve their timeout issue.